modules/social_features/social_profile/src/SocialProfileRenderService.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Drupal\social_profile;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\profile\Entity\ProfileInterface;
+use Drupal\user\Entity\User;
+use Drupal\user\UserInterface;
+
+/**
+ * Class SocialProfileRenderService.
+ *
+ * @package Drupal\social_profile
+ */
+class SocialProfileRenderService {
+
+  /**
+   * The Profile Entity Storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $profileStorage;
+
+  /**
+   * Cached profiles.
+   *
+   * @var \Drupal\user\UserInterface[]|\Drupal\Core\Session\AccountInterface[]
+   */
+  protected $userProfileCache = [];
+
+  /**
+   * The Profile View Builder.
+   *
+   * @var \Drupal\Core\Entity\EntityViewBuilderInterface
+   */
+  protected $profileViewBuilder;
+
+  /**
+   * SocialProfileRenderService Constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The Entity Type Manager.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->profileStorage = $entity_type_manager->getStorage('profile');
+    $this->profileViewBuilder = $entity_type_manager->getViewBuilder('profile');
+  }
+
+  /**
+   * Renders the profile of a user
+   *
+   * @param \Drupal\user\UserInterface|\Drupal\Core\Session\AccountInterface $account
+   *   The user account for which we render the profile.
+   * @param $view_mode
+   *   The view mode in which the profile gets rendered.
+   *
+   * @return array
+   *   A render array for the profile.
+   */
+  public function renderUserProfile($account, $view_mode) {
+    // Build cache ID.
+    $cid = $account->id() . ':' . $view_mode;
+
+    // Check if we already have rendered something previously.
+    if (!empty($this->userProfileCache[$cid])) {
```

Comment:
We should use `isset` here instead of `empty`. It could be that in an earlier call we've already determined that this would yield an invalid render and set the cache value to an empty array (`[]`). When using `empty` this would still result in recalculating the result.